modules/roms.py

```diff
@@ -123,27 +123,29 @@ def list_available_roms(force_recheck: bool = False) -> list[ROM]:
     :return: List of all the valid ROMS that have been found
     """
     global rom_cache
-    if force_recheck or len(rom_cache) == 0:
-        if not ROMS_DIRECTORY.is_dir():
-            raise RuntimeError(f"Directory {str(ROMS_DIRECTORY)} does not exist!")
 
+    if force_recheck:
         rom_cache.clear()
-        for file in ROMS_DIRECTORY.iterdir():
-            if file.is_file():
+
+    if not ROMS_DIRECTORY.is_dir():
+        raise RuntimeError(f"Directory {str(ROMS_DIRECTORY)} does not exist!")
+
+    for file in ROMS_DIRECTORY.iterdir():
+        if file.is_file():
+            if force_recheck or str(file) not in rom_cache:
                 try:
-                    rom_cache.append(load_rom_data(file))
+                    rom_cache[str(file)] = load_rom_data(file)
```

Comment:
you could just call load_rom_data(file) here, because that already adds the rom to the cache.